bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/product/VariantValue.java

```diff
@@ -25,9 +27,17 @@
 public interface VariantValue {
     String getLabel();
 
+    default String getDefaultLabel() {
+        return null;
+    }
+
     Integer getId();
 
     default String getUid() {
         return null;
     }
+
+    default SwatchDataInterface getSwatchData() {
```

Comment:
We should not add a dependency to the SwatchDataInterface (DTO) here. This would couple the API to closely to the gql binding and if we have to make changes we cannot as we don't own the interface.

Ideally the `Variant` interface defines it's type, for example as an enum: `COLOR`, `TEXT`. This could be set in the model and be read in the htl script accordingly without exposing the DTO.